File: QE-Java Selenium/Activity8.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity8 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/mouse-events");
        driver.manage().window().maximize();

        System.out.println(driver.getTitle());

        Actions builder = new Actions(driver);
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        WebElement cargoLock = driver.findElement(By.cssSelector(".file:nth-of-type(1)"));

        WebElement cargoToml = driver.findElement(By.cssSelector(".file:nth-of-type(2)"));

        WebElement src = driver.findElement(By.cssSelector(".file:nth-of-type(3)"));

        WebElement target = driver.findElement(By.cssSelector(".file:nth-of-type(4)"));

        builder.click(cargoLock)
               .moveToElement(cargoToml)
               .click()
               .build()
               .perform();

        System.out.println(driver.findElement(By.id("result")).getText());

        builder.doubleClick(src)
               .contextClick(target)
               .build()
               .perform();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("li.w-full:nth-of-type(1)")));

        WebElement openButton = driver.findElement(By.cssSelector("li.w-full:nth-of-type(1)"));

        builder.click(openButton).build().perform();

        System.out.println(driver.findElement(By.id("result")).getText());

        driver.quit();
    }
}

File: QE-Java Selenium/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/mouse-events")
driver.maximize_window()

print(driver.title)

actions = ActionChains(driver)
wait = WebDriverWait(driver, 10)

cargo_lock = driver.find_element(By.CSS_SELECTOR, ".file:nth-of-type(1)")

cargo_toml = driver.find_element(By.CSS_SELECTOR, ".file:nth-of-type(2)")

src = driver.find_element(By.CSS_SELECTOR, ".file:nth-of-type(3)")

target = driver.find_element(By.CSS_SELECTOR, ".file:nth-of-type(4)")

actions.click(cargo_lock) \
       .move_to_element(cargo_toml) \
       .click() \
       .perform()

print(driver.find_element(By.ID, "result").text)

actions.double_click(src) \
       .context_click(target) \
       .perform()

wait.until(EC.visibility_of_element_located((By.CSS_SELECTOR, "li.w-full:nth-of-type(1)")))

open_button = driver.find_element(By.CSS_SELECTOR, "li.w-full:nth-of-type(1)")

actions.click(open_button).perform()

print(driver.find_element(By.ID, "result").text)

driver.quit()








